types/src/crypto/asymmetric_key.rs

```diff
@@ -209,8 +213,8 @@ impl Clone for SecretKey {
     fn clone(&self) -> Self {
         match self {
             SecretKey::System => SecretKey::System,
-            SecretKey::Ed25519(sk) => Self::ed25519(*sk.as_bytes()),
-            SecretKey::Secp256k1(sk) => Self::secp256k1(*sk),
+            SecretKey::Ed25519(sk) => Self::ed25519(*sk.as_bytes()).unwrap(),
```

Comment:
If we're being strict about not unwrapping in constructors, then I think we should be consistently zealous here.  We should remove the `Clone` impl and provide something like `try_clone(&self) -> Result<Self, Error>` I guess?

Reply:
I agree that `SecretKey` should not have `Clone`, but it will add more scope to an already complex PR.

I am nervous about `try_clone`.  We really should never be cloning a private key.